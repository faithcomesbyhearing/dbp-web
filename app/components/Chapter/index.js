/**
 *
 * Chapter
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import Link from 'next/link';

const Chapter = ({ as, active, chapter, clickHandler, href }) => {
	if (active) {
		return (
			<a className={'chapter-box'} onClick={clickHandler}>
				<span className={'active-chapter'}>{chapter}</span>
			</a>
		);
	}
	return (
		<Link legacyBehavior href={href} as={as}>
			<a className={'chapter-box'} onClick={clickHandler}>
				<span>{chapter}</span>
			</a>
		</Link>
	);
};

Chapter.propTypes = {
	chapter: PropTypes.number,
	active: PropTypes.bool,
	href: PropTypes.string,
	as: PropTypes.string,
	clickHandler: PropTypes.func,
};

export default Chapter;
