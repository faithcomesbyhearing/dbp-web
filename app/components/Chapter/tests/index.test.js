import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import Chapter from '..';

describe('<Chapter />', () => {
	const clickMessage = 'click-handler';
	const clickHandler = jest.fn(() => clickMessage);

	it('Should render and display chapter number', () => {
		render(<Chapter clickHandler={clickHandler} chapter={1} active href={''} as={''} />);
		expect(screen.getByText('1')).toBeInTheDocument();
	});

	it('Should render for active chapter', () => {
		render(<Chapter clickHandler={clickHandler} chapter={1} active href={''} as={''} />);
		expect(screen.getByText('1')).toHaveClass('active-chapter');
	});

	it('Should render for inactive chapter', () => {
		render(<Chapter clickHandler={clickHandler} chapter={1} href={''} as={''} />);
		expect(screen.getByText('1')).not.toHaveClass('active-chapter');
	});

	it('Should match snapshot for active chapter', () => {
		const { asFragment } = render(<Chapter clickHandler={clickHandler} chapter={1} active href={''} as={''} />);
		expect(asFragment()).toMatchSnapshot();
	});

	it('Should match snapshot for inactive chapter', () => {
		const { asFragment } = render(<Chapter clickHandler={clickHandler} chapter={1} href={''} as={''} />);
		expect(asFragment()).toMatchSnapshot();
	});

	it('Should correctly fire click event', () => {
		const { container } = render(<Chapter clickHandler={clickHandler} chapter={1} active href={''} as={''} />);
		// const anchor = screen.getByRole('link', { name: '1' });
		const anchor = container.querySelector('.chapter-box');
		fireEvent.click(anchor);
		expect(clickHandler).toHaveBeenCalledTimes(1);

		const clickResult = clickHandler.mock.results[0].value;
		expect(clickResult).toEqual(clickMessage);
	});
});
