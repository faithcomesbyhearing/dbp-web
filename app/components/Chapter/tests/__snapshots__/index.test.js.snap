// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Chapter /> Should match snapshot for active chapter 1`] = `
<DocumentFragment>
  <a
    class="chapter-box"
  >
    <span
      class="active-chapter"
    >
      1
    </span>
  </a>
</DocumentFragment>
`;

exports[`<Chapter /> Should match snapshot for inactive chapter 1`] = `
<DocumentFragment>
  <a
    class="chapter-box"
    href=""
  >
    <span>
      1
    </span>
  </a>
</DocumentFragment>
`;
